packages/image_picker/image_picker/example/lib/main.dart

```diff
@@ -114,10 +121,11 @@ class _MyHomePageState extends State<MyHomePage> {
   }
 
   Future<void> _disposeVideoController() async {
-    if (_controller != null) {
-      await _controller.dispose();
-      _controller = null;
+    if (_toBeDisposed != null) {
```

Comment:
I don't get this. Are we expecting `_disposeVideoController` to be called twice and only disposing the controller the second time?

Reply:
The issue is that we **`dispose` of the controller before we're done with it.**

The situation is the following: 

You pick a video and while it's playing, you pick a second video.

The first thing the playVideo method does is to [call `_disposeVideoController`](https://github.com/flutter/plugins/blob/master/packages/image_picker/image_picker/example/lib/main.dart#L54) which disposes of the current controller.

Then we create a [new video controller and cause a re-render](https://github.com/flutter/plugins/blob/master/packages/image_picker/image_picker/example/lib/main.dart#L55-L64), which ends up [updating the VideoPlayer widget](https://github.com/flutter/plugins/blob/master/packages/image_picker/image_picker/example/lib/main.dart#L388) with the newest controller (through the `_previewVideo` method). Then we get here:

https://github.com/flutter/plugins/blob/master/packages/video_player/video_player/lib/video_player.dart#L537

That is attempting to remove a listener from the *old* controller.

The old controller has already been disposed of, and that `removeListener` [throws an Assertion](https://api.flutter.dev/flutter/foundation/ValueNotifier/removeListener.html). 

In Safari the assertion is fatal (it interrupts all Javascript, and halts the program). In every other platform, it gets logged (but I don't think we should ignore an assertion?).

This is to dispose of the controller later, when we're really sure that nobody else is using it anymore.

(Note that the LAST controller will get taken care of by the [`dispose` method of the _MyHomePageState class](https://github.com/flutter/plugins/blob/master/packages/image_picker/image_picker/example/lib/main.dart#L109))